### ISSUE TITLE
problems with update-claim in batch mode

### ISSUE THREAD

**Issue by rwst** [maxlath/wikibase-cli#129]
With
`{"guid":"Q417654$8a1313ea-74ee-4ed5-a7ed-90c8aae09762","newValue":"InChi=1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+"}`
in `tmp` and
`wd uc -b <tmp` I get:
```
edit group: https://tools.wmflabs.org/editgroups/b/wikibase-cli/4ba4bc7ea38ad/
processing line 1: { "guid": "Q417654$8a1313ea-74ee-4ed5-a7ed-90c8aae09762", "newValue": "InChi=1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+" }
old and new claim values can't be the same { oldValue: undefined, newValue: undefined }
```

A similar attempt with the line
`Q417654$8A1313EA-74EE-4ED5-A7ED-90C8AAE09762 InChi=1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+` gives
```
edit group: https://tools.wmflabs.org/editgroups/b/wikibase-cli/d2e5237510b47/
processing line 1: Q417654$8A1313EA-74EE-4ED5-A7ED-90C8AAE09762 InChi=1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+
missing snak value { property: 'P234', value: undefined }
```

The format is the same as in the documentation so I'm quite surprised. Is batch uc possible at all?

**Comment by rwst** [comment]
Note the documentation says only the new value is needed. I tried with old and new value. First simple format:
```
ralf@ark:~/wikidata> wd uc -b -v <tt
edit group: https://tools.wmflabs.org/editgroups/b/wikibase-cli/a6a5371ff938a/
processing line 1: Q417654$8A1313EA-74EE-4ED5-A7ED-90C8AAE09762 1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+ InChi=1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+
invalid monolingualtext value { property: 'P1476', value: 'triprolidine' }
```
Note the statement is P234 not P1476. The other format:
```
ralf@ark:~/wikidata> wd uc -b -v <tt
edit group: https://tools.wmflabs.org/editgroups/b/wikibase-cli/d18a28a291187/
processing line 1: { "guid": "Q417654$8A1313EA-74EE-4ED5-A7ED-90C8AAE09762", "oldValue": "1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+", "newValue": "InChi=1S/C19H22N2/c1-16-7-9-17(10-8-16)18(19-6-2-3-12-20-19)11-15-21-13-4-5-14-21/h2-3,6-12H,4-5,13-15H2,1H3/b18-11+" }
old and new claim values can't be the same { oldValue: undefined, newValue: undefined }
```

**Comment by rwst** [comment]
Actually, because of P1476 seems related to #128 . But note P1476 is not used by these items, at all.

**Comment by maxlath** [comment]
which version of `wikibase-cli` are you using?

**Comment by rwst** [comment]
The P1476 comes from the reference (which apparently gets rewritten)! 

```
ralf@ark:~/wikidata> wd --version
12.0.3
```

**Comment by rwst** [comment]
Confirmed this happens with 15.2.1 too.

**Comment by rwst** [comment]
I confirm uc batch works on items that don't have P1476 but only when I give four arguments: **guid property oldValue newValue** which needs to be changed in the documentation.

**Comment by rwst** [comment]
No sorry that's wrong. The two arguments work fine, as in docs. All the problems come from P1476.

**Comment by maxlath** [comment]
There were 2 problems here:
- one due to the wikibase-edit internal claim serialization, as identified in #128
- one due to differences in case between the provided guid and the one returned by the Wikibase API, [this should not be a problem anymore](https://github.com/maxlath/wikibase-edit/commit/79c88067)

Both problems have now been patched and published in `v15.2.2`, and [I was able to do the edit you provided](https://www.wikidata.org/w/index.php?title=Q417654&diff=prev&oldid=1278775001)

**Comment by rwst** [comment]
It works. Many thanks!
